packages/amplify_datastore/example/integration_test/query_test/pagination_test.dart

```diff
@@ -0,0 +1,112 @@
+/*
+ * Copyright 2021 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+import 'package:amplify_datastore/amplify_datastore.dart';
+
+import 'package:amplify_datastore_example/models/ModelProvider.dart';
+import 'package:integration_test/integration_test.dart';
+import 'package:flutter_test/flutter_test.dart';
+import 'package:amplify_flutter/amplify.dart';
+import 'package:collection/collection.dart';
+
+import '../utils/setup_utils.dart';
+
+void main() {
+  IntegrationTestWidgetsFlutterBinding.ensureInitialized();
+
+  group('pagination', () {
+    var models = List.generate(1000, (i) => Blog(name: 'blog $i'));
+    setUpAll(() async {
+      await configureDataStore();
+      await clearDataStore();
+      for (var model in models) {
+        await Amplify.DataStore.save(model);
+      }
+    });
+    testWidgets('should return the models for the given page and limit',
+        (WidgetTester tester) async {
+      // page 0
+      var pageZeroBlogs = await Amplify.DataStore.query(Blog.classType,
+          pagination: QueryPagination(page: 0, limit: 10));
+      var expectedPageZeroBlogs = models.getRange(0, 10).toList();
```

Comment:
Should we add a sorting constrain to the query while comparing two lists using `ListEquality().equals` which the order of item matters for.

Reply:
This is a good point. I updated the tests to use unorderedEquals and orderedEquals. See my comment above which is sort of related as well